tv_apps/smart-home/js/edit.js

```diff
@@ -409,24 +409,47 @@
     },
 
     openDeleteCardDialog: function(scrollable, nodeElem) {
-      Utils.holdFocusForAnimation();
-      this.modalDialog.open({
-        message: {
-          textL10nId: 'delete-card-alert'
-        },
-        buttonSettings: [
-          {
-            textL10nId: 'cancel',
-            defaultFocus: true,
-            onClick: this.spatialNavigator.focus.bind(this.spatialNavigator)
+      return new Promise((resolve, reject) => {
```

Comment:
This is to let outside caller know the right timing that deleting card job ends.
The code has been changed to use the event to reduce the complexity.